vignettes/opengraph.Rmd

```diff
@@ -0,0 +1,106 @@
+[Open Graph protocol][ogp] for rich social media cards.
+In addition to specifying an alternate description for the package
+and any individual articles,
+you may also choose the preview image shown 
+and the style of card used on Twitter.
+
+You can preview and validate the appearance of the social media card previews with 
+
+- [Twitter Card Validator][twitter-validator]
+
+- [Google Structured Data Testing Tool][ogp-validator]
+
+## Site-wide customizations
+
+Metadata for the entire pkgdown website can be specified 
+in the site's `_pkgdown.yml` configuration file in the 
+`home` and `template: opengraph` sections.
+
+```yaml
+home:
+  title: An R package for pool-noodle discovery
+  description: Discover and add pool-noodles to your growing collection.
+  
+template:
+  opengraph:
+    image: 
+      src: man/figures/card.png
+      alt: "Pool noodles configured to form the word poolnoodlr"
+    twitter: 
+      creator: "@hadleywickham"
+      site: "@rstudio"
+      card: summary_large_image
+```
+
+The `home: title` and `home: description` fields override 
+the `Title` and `Description` fields in the package `DESCRIPTION`.
+As mentioned in the `build_home()` documentation,
+it's good practice to change these fields to make your package documentation
+easier to find via search engines.
+
+The `template: opengraph` section allows you to further customize the social media card preview.
+
+By default, pkgdown uses the package's logo for the card preview (if one exists).
+Custom preview images can be stored in `man/figures/` 
+alongside other images used in the package documentation,
+as recommended in the `build_home()` documentation.
+The `src` field is required and can alternatively be a fully qualified URL,
+and it is highly recommended that you include the `alt` field for screen readers.
+
+The Twitter handle of the creator and their organization 
+can be specified in `twitter: creator` and `twitter: site` fields.
+If only one of these two fields is specified, the other will be filled in.
+The `twitter: card` option allows the user 
+to specify the style of card displayed on Twitter,
+chosen from the options [listed here][twitter-card].
+For pkgdown sites, the relevant options are `summary_large_image` and `summary`.
+The latter option features a small image inline with the package title and description.
+
+
+## Article metadata
+
+Articles and vignettes will use the metadata specified in the site's `_pkgdown.yml`,
+or metadata specific to the article can be provided in the article's YAML header.
+
+```markdown
+---
+author: "Mara Averick"
+title: "Introduction to poolnoodlr"
+description: "A brief introduction to pool noodles in R."
+opengraph:
+  image: 
+    src: "http://example.com/pkg/batpig.png"
+  twitter:
+    card: summary
+    creator: "@dataandme"
+---
+```
+
+Use the `author`, `title`, and `description` fields 
+to specify the author, title, and description of the vignette or article.
```

Comment:
Needs to explain why you'd want to do this rather than relying on the metadata provided in the vignette.

Reply:
I updated this section to expand on the top-level YAML fields used in vignettes.

In doing this, I also realized that articles without a top-level `description` have no description at all. I updated `build_article()` to use the package name as the default description for articles without a description. This commit would be easy to revert if you prefer the previous behavior.